### PR TITLE
Error out when locallock and proclock are not consistent. 

### PR DESCRIPTION
**Pull request by zhangh43**
When acquire a lock, there is a case where locallock doesn't hold
the lock but global lock treats we already held the lock. This
inconsistent state may be caused by broken of locallock or
global lock. We dump their details in the log here to help analysis.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [x] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/backend/storage/lmgr/lock.c [hunk 1]
```diff
@@ -1006,10 +1006,31 @@ LockAcquireExtended(const LOCKTAG *locktag,
 	{
 		if (proclock->holdMask & LOCKBIT_ON(lockmode))
 		{
-			elog(LOG, "lock %s on object %u/%u/%u is already held",
+			elog(LOG, "lock %s on object %u/%u/%u%hu type %hhu is already held",
```
Comment (asimrp) [inline src/backend/storage/lmgr/lock.c]: Should we have a space separating the `%u` and `%hu`?

Comment (zhangh43) [reply src/backend/storage/lmgr/lock.c]: fixed

**Comment by zhangh43** [comment]
@asimrp yes, I'll discuss it in upstream as well

**Comment by zhangh43** [comment]
I checked the upstream code. Upstream elog error when locallock and global lock are not consistent. Since locallock and global lock may be broken, It's better to error out as early as possible. So changed this PR from dump more logs to make it consistent with upstream.

**Pull request merged**